src/apps/Hosting/Files.js

```diff
@@ -89,12 +118,12 @@ const HostingFilesView = React.createClass({
         <InnerToolbar title="Website Hosting">
           <Show if={items.length && !isLoading}>
             <RaisedButton
-              label={isDefaultHosting ? 'Published' : 'Publish'}
-              onTouchTap={this.handleShowPublishDialog}
+              label="Go to site"
               primary={true}
-              disabled={isDefaultHosting}
-              disabledBackgroundColor={Colors.green500}
-              disabledLabelColor="#FFF"
+              icon={<FontIcon className="synicon-open-in-new" style={{ marginTop: 4 }} />}
+              onTouchTap={!hasHostingUrl && this.showSnackbar}
```

Comment:
Created handler as suggested. And renamed `showSnackbar` method. When i moved `setSnackbarNotification` inside of handler it was triggered when user entered page. So let's leave it as it is.